File: 0977_Squares_of_ a_Sorted_Array/Solution.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        int l = 0;
        int r = nums.length - 1;
        int[] res = new int[nums.length];
        int i = r;
        while (l <= r) {
            if (Math.abs(nums[l]) <= Math.abs(nums[r])) {
                res[i] = nums[r] * nums[r];
                r--;
            } else {
                res[i] = nums[l] * nums[l];
                l++;
            }
            i--;
        }
        return res;
    }
}

File: 0977_Squares_of_ a_Sorted_Array/Solution.cpp
#include <vector>

using namespace std;

class Solution
{
public:
    vector<int> sortedSquares(vector<int> &nums)
    {
        int l = 0;
        int r = nums.size() - 1;
        vector<int> res(nums.size());
        int i = r;
        while (l <= r)
        {
            if (abs(nums[l]) <= abs(nums[r]))
            {
                res[i] = nums[r] * nums[r];
                --r;
            }
            else
            {
                res[i] = nums[l] * nums[l];
                ++l;
            }
            --i;
        }
        return res;
    }
};
